### ISSUE TITLE
Related lists not working

### ISSUE THREAD

**Issue by ihubsys** [sardoj/VTModuleDesigner#21]
Hi there,

firstly I want the thank you for the fantastic module designer - it really makes it so easy for a non programmer like me to create new modules.

I'm having a issue with making the related lists work in module designer for existing modules. Here is what I have done so far.
1. In project tasks I have added a related field "Project milestones" (I want to related tasks to milestones)
2. In project milestones I have added a related list -"Project Tasks"

Then I go into a project task and Add a related milestone - this part works fine. BUT when I go to project milestones and click the newly added Project tasks tab the list is empty.

A few other observations:

Re: Adding project task from project milestone 
when I am view the related project task list in Project milestone I can add a project task from the list view - in this case the new project task is automatically related to the project milestone - so this works as it should.

Re: Selecting project tasks
If I try and select a project task in the related list view, it is added to the list BUT the Related to project milestone field in project tasks is not updated.

I hope that all makes sense.

any suggestion on how to get that all working?

Mark

**Comment by jayo2005** [comment]
I have same problem did you get it sorted?

**Comment by nelcab** [comment]
Hi,

I got into something similar.  When I use a related field from a custom module in another custom module, the Popup Window doesn't show the list.

I discovered a way to fix it, changing manually the "your-module".php file in the related module (the one you're quering information).  

I didn't have the time to make a patch to ModuleDesigner, but maybe later on I could work on that.

In the meantime, I leave you where I found a clue what was happening: 
http://vtiger-crm.2324883.n4.nabble.com/Relate-two-custom-modules-td4646.html
Check the arrays for "Search Fields"

Hope it helps, good luck.

**Comment by alexanderjp** [comment]
Has anyone received the solution for this?

I am having the same problem when I create an item in the child module, I do not see it in the list of the parent module.  However in the parent module I can select the child item no problem.

**Comment by playmono** [comment]
Hi all,

I look a little on to this and didn't understand the problem.

> I'm having a issue with making the related lists work in module designer for existing modules. Here is what I have done so far.
> 1. In project tasks I have added a related field "Project milestones" (I want to related tasks to milestones)
> 2. In project milestones I have added a related list -"Project Tasks"
> 
> Then I go into a project task and Add a related milestone - this part works fine. BUT when I go to project milestones and click the newly added Project tasks tab the list is empty."

I find nothing wrong with this. I did a related list between Project Tasks and Project Milestones and viceversa without problems.

> In the meantime, I leave you where I found a clue what was happening: 
> http://vtiger-crm.2324883.n4.nabble.com/Relate-two-custom-modules-td4646.html

This post is from 2011, so I doubt this is the same problem as you described. I think in 2011 they were using vTiger 5.1.

> Has anyone received the solution for this?
> 
> I am having the same problem when I create an item in the child module, I do not see it in the list of the parent module. However in the parent module I can select the child item no problem.

Can you be a little more especific?

Thank you all.

**Comment by alexanderjp** [comment]
You are right, the issue is a little different.  It is actually with the ability to select records from a child module.  Here is the example I have in place.  

I created a module call "PROPERTY" and another called "TELEPHONE NUMBERS"

WORKING
When in the PROPERTY module I can go to the related list and select "+ Add Telephone Number" button.  This will create a new item in the TELEPHONE NUMBER module and list in the PROPERTY module.

NOT WORKING
If i create an item in the TELEPHONE NUMBER module then go to PROPERTY module related lists and click "Select TELEPHONE NUMBER", I see the number I want to associated and select it.  The window closes but does not associate to the PROPERTY module.

WORKING
I can go to the TELEPHONE NUMBER module and associate the item to PROPERTY module and it will list under the PROPERTY modules' related lists.

Does that make sense?

**Comment by alexanderjp** [comment]
Here is some more information.

It works when I set the method as get_related_list() in the PROPERTY module "Related Lists" settings but the Add button will not update the child item.

If is set the method as  get_dependents_list() the select does not work but the add does.
